app/code/Magento/InventoryCatalog/Model/IsBackOrderAvailable.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Model;
```

Comment:
@maghamed: shouldn't this class belong to `InventoryConfiguration` module just like `IsNotManageStock`?